File: ChatPlays/ChatPlays/Overlord.js
let clientId = "lq1tzm68ep4qvipym40apgizf45sr6";
let clientSecret = "22i9hawqpws4s4k3q99via9vtywbr0";

export default {}

export function initTwitchConnections()
{

}

function getTwitchAuth() {
    let url = 'https://id.twitch.tv/oauth2/token?client_id=$lq1tzm68ep4qvipym40apgizf45sr6&client_secret=$22i9hawqpws4s4k3q99via9vtywbr0&grant_type=client_credentials';
    fetch((url), {
        method: "POST",
    })
        .then((res) => res.json())
        .then((data) => handleAuth(data));
}

function handleAuth(data) {
    let { access_token, expires_in, token_type } = data;
    //console.log('${token_type} ${access_token}');
}

getTwitchAuth();

File: ChatPlaysBG3/ChatPlaysBG3/Overlord.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;
using TwitchLib.Api;
using TwitchLib.Api.Auth;
using TwitchLib.Api.Core.Exceptions;
using TwitchLib.Api.Helix.Models.Bits;

namespace ChatPlaysBG3
{
    internal class Overlord
    {
        private Archive localArchive;

        public event EventHandler OnAuthRecieved;

        public Overlord(Archive archive) 
        {
            localArchive = archive;
            getAuth();
        }

        internal async void getAuth()
        {
            using var httpClient = new HttpClient();
            var tokenUrl = "https://id.twitch.tv/oauth2/token";
            var content = new FormUrlEncodedContent(new[]
            {
                new KeyValuePair<string, string>("client_id", localArchive.ClientId),
                new KeyValuePair<string, string>("client_secret", localArchive.ClientSecret),
                new KeyValuePair<string, string>("grant_type", "client_credentials")
            });

            try
            {
                var responce = await httpClient.PostAsync(tokenUrl, content);

                if (responce.IsSuccessStatusCode)
                {
                    var jsonContent = await responce.Content.ReadAsStringAsync();
                    var jObject = JObject.Parse(jsonContent);
                    var auth = (string)jObject["access_token"];
                    localArchive.Authorization = $"Bearer {auth}";
                    OnAuthRecieved(this, EventArgs.Empty);
                }
                else
                {
                    Console.WriteLine($"Error: {responce.StatusCode}");
                }
            }
            catch (Exception ex) { Console.WriteLine($" Error, auth Exception: {ex.Message}"); }


        }

    }
}
